pkg/testing/integration_test.go

```diff
@@ -353,6 +422,106 @@ func (s *IntegrationSuite) createProfile(c *C, ctx context.Context) string {
+	}
+	s.repositoryServer.repositoryServer.Spec.Storage.CredentialSecretRef = v1.SecretReference{
+		Name:      s3Creds.GetName(),
+		Namespace: s3Creds.GetNamespace(),
+	}
+	s.repositoryServer.repositoryServer.Spec.Repository.PasswordSecretRef = v1.SecretReference{
+		Name:      repositoryPassword.GetName(),
+		Namespace: repositoryPassword.GetNamespace(),
+	}
+	s.repositoryServer.repositoryServer.Spec.Server.UserAccess.UserAccessSecretRef = v1.SecretReference{
+		Name:      userAccess.GetName(),
+		Namespace: userAccess.GetNamespace(),
+	}
+	s.repositoryServer.repositoryServer.Spec.Server.AdminSecretRef = v1.SecretReference{
+		Name:      serverAdminUser.GetName(),
+		Namespace: serverAdminUser.GetNamespace(),
+	}
+	s.repositoryServer.repositoryServer.Spec.Server.TLSSecretRef = v1.SecretReference{
+		Name:      tls.GetName(),
+		Namespace: tls.GetNamespace(),
+	}
+
+	// Create RepositoryServer CR
+	repositoryServer, err := s.crCli.RepositoryServers(testutil.DefaultKanisterNamespace).Create(ctx, s.repositoryServer.repositoryServer, metav1.CreateOptions{})
+	c.Assert(err, IsNil)
+
+	// Wait for controller to set pod name field
+	timeoutCtx, waitCancel := context.WithTimeout(ctx, contextWaitTimeout)
+	defer waitCancel()
+	err = poll.Wait(timeoutCtx, func(ctx context.Context) (bool, error) {
+		rs, err := s.crCli.RepositoryServers(testutil.DefaultKanisterNamespace).Get(ctx, repositoryServer.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		switch {
+		case rs.Status.ServerInfo.PodName != "":
+			contentCacheMB, metadataCacheMB := kopiacmd.GetGeneralCacheSizeSettings()
+			commandArgs := kopiacmd.RepositoryCommandArgs{
+				CommandArgs: &kopiacmd.CommandArgs{
+					RepoPassword:   testutil.DefaultRepositoryPassword,
+					ConfigFilePath: kopiacmd.DefaultConfigFilePath,
+					LogDirectory:   kopiacmd.DefaultLogDirectory,
+				},
+				CacheDirectory:  kopiacmd.DefaultCacheDirectory,
+				Hostname:        testutil.DefaultRepositoryServerHost,
+				ContentCacheMB:  contentCacheMB,
+				MetadataCacheMB: metadataCacheMB,
+				Username:        testutil.DefaultKanisterAdminUser,
+				RepoPathPrefix:  testutil.DefaultRepositoryPath,
+				Location:        s3Location.Data,
+			}
+			err = repository.ConnectToOrCreateKopiaRepository(
+				s.cli,
+				testutil.DefaultKanisterNamespace,
+				rs.Status.ServerInfo.PodName,
+				testutil.DefaultKopiaRepositoryServerContainer,
+				commandArgs,
+			)
+			if err != nil {
+				return false, nil
+			}
+			return true, nil
+		}
+		return false, nil
+	})
+	c.Assert(err, IsNil)
+
+	// Wait for the Repository Server to Be in Ready Stage.
+	timeoutCtx, waitCancel = context.WithTimeout(ctx, contextWaitTimeout)
+	defer waitCancel()
+	err = poll.Wait(timeoutCtx, func(ctx context.Context) (bool, error) {
+		rs, err := s.crCli.RepositoryServers(testutil.DefaultKanisterNamespace).Get(ctx, repositoryServer.Name, metav1.GetOptions{})
+		if rs.Status.Progress == "ServerReady" && err == nil {
```

Comment:
I think there should be a const for this value `ServerReady`, let's use that.

Reply:
Done !